src/main/java/gov/cabinetoffice/gap/lambda/Handler.java

```diff
@@ -79,50 +82,55 @@ public SQSBatchResponse handleRequest(final SQSEvent event, final Context contex
+                        logger.info("Finished fetching completedGrantExports with size of: {}", completedGrantExports.getGrantExports().size());
 
-                    final String superZipFilename = HelperUtils.generateFilename(submission.getSchemeName(), "");
+                        ZipService.createSuperZip(completedGrantExports.getGrantExports());
 
-                    final String superZipObjectKey = ZipService.uploadZip(submission.getSchemeId() + "/" + exportBatchId, superZipFilename);
+                        final String superZipFilename = HelperUtils.generateFilename(submission.getSchemeName(), "");
 
-                    GrantExportBatchService.addS3ObjectKeyToGrantExportBatchRecord(restClient, exportBatchId, superZipObjectKey);
-                    GrantExportBatchService.updateGrantExportBatchRecordStatus(restClient, exportBatchId, GrantExportStatus.COMPLETE);
+                        final String superZipObjectKey = ZipService.uploadZip(submission.getSchemeId() + "/" + exportBatchId, superZipFilename);
 
-                    NotifyService.sendConfirmationEmail(restClient, emailAddress, exportBatchId, submission.getSchemeId(),
-                            submissionId);
-                } catch (Exception e) {
-                    logger.error("Could not process message while trying to create super zip", e);
-                    GrantExportBatchService.updateGrantExportBatchRecordStatus(restClient, exportBatchId, GrantExportStatus.FAILED);
+                        GrantExportBatchService.addS3ObjectKeyToGrantExportBatchRecord(restClient, exportBatchId, superZipObjectKey);
+                        GrantExportBatchService.updateGrantExportBatchRecordStatus(restClient, exportBatchId, GrantExportStatus.COMPLETE);
+                    } catch (Exception e) {
+                        logger.error("Could not process message while trying to create super zip", e);
+                        GrantExportBatchService.updateGrantExportBatchRecordStatus(restClient, exportBatchId, GrantExportStatus.FAILED);
+                    }
                 }
+
+                NotifyService.sendConfirmationEmail(restClient, emailAddress, exportBatchId, submission.getSchemeId(),
+                        submissionId);
             } else {
                 logger.info(
-                        String.format("Outstanding exports for export batch %s: %s", exportBatchId, outstandingCount));
+                        "Outstanding exports for export batch {}: {}", exportBatchId, outstandingCount);
             }
 
             // STEP 9 - clear tmp dir as this is preserved between frequent invocations
             ZipService.deleteTmpDirContents();
             logger.info("Tmp dir cleared");
         } catch (Exception e) {
             logger.error("Could not process message", e);
-            ExportRecordService.updateExportRecordStatus(restClient, exportBatchId, submissionId, GrantExportStatus.FAILED);
+
```

Comment:
Before the super zip work we were throwing a RuntimeException here. I'm not sure if we want to add that back or not